gp-includes/routes/translation.php

```diff
@@ -265,11 +265,13 @@ public function translations_post( $project_path, $locale_slug, $translation_set
 					$data[ "translation_$i" ] = $translations[ $i ];
 				}
 			}
+			$set_status = $data['status'];
+			$data['status'] = 'waiting';
 
 			if ( $this->can( 'approve', 'translation-set', $translation_set->id ) || $this->can( 'write', 'project', $project->id ) )
-				$data['status'] = 'current';
```

Comment:
Yes. Further down `set_status( 'current' )` can actually fail which it could not do before. So if it does, you'd end up with two current translations of which one should not be current in the first place.